### ISSUE TITLE
Consider replace swig template engine

### ISSUE THREAD

**Issue by Haocen** [iissnan/hexo-theme-next#1892]
### I agree and want to create new issue 

- [x] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [x] Yes, I was on [NexT Documentation Site](http://theme-next.iissnan.com/).
- [x] And yes, I already searched for current [issues](https://github.com/iissnan/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Steps to reproduce the behavior 
1. N/A
2. N/A
3. N/A

* Link to demo site with this issue: N/A
* Link(s) to source code or any usefull link(s): N/A

### NexT Information

**NexT Version:**

```
[x] Latest Master branch.
[x] Latest Release version.
[x] Old version - 
```

**NexT Scheme:**

```
[x] All schemes
[] Muse
[] Mist
[] Pisces
[] Gemini
```

### Other Information 

According to https://github.com/paularmstrong/swig/

swig template engine is no longer maintained, we should consider switching to another template engine.

ejs seemed has similar syntax and is going to stick around for a while, I propose to use ejs as a replacement.

**Comment by ivan-nginx** [comment]
Yeah, i know it. But it's hardly and need very big time to refactoring from swig to ejs and at most cases swig for now work fine. And there is one thing anyway: hexo also use swig as default template and as i know they not want for now to migrate to any another.
If u can and want, u may do it in separated test branch. Or anyone? Who want to do it, please, welcome!

**Comment by Neveryu** [comment]
oh,no.wish you can do it finish..

**Comment by AlynxZhou** [comment]
@ivan-nginx Recently I found nunjucks's syntax is similiar to swig, if we have plan to migrate in future, I think nunjuck is a better choice than ejs. XD

**Comment by ivan-nginx** [comment]
There are many rendering engines, not only `ejs`, `jade` or `swig`. For example, `mustashe` have similar syntax too. But i don't know why we must use **mainstream** rendering. Swig is work fine and no have any critical bugs. And remake rendering engine will make some special troubles and additional bugs in future.

Anyway, if there will be strong arguments about replace template engine, we can try to implement it. But question is still open: who will do it? It will take a lot of time and i'm not shure what anybody will do it for free.

**Comment by ivan-nginx** [comment]
Already replaced `swig` by `swig-templates` in [this pull](https://github.com/hexojs/hexo/pull/2949) and released in [Hexo v3.5.0](https://github.com/hexojs/hexo/releases/tag/3.5.0) and tested with [NexT v6](https://github.com/theme-next/hexo-theme-next).
